### ISSUE TITLE
error: mismatched types:

### ISSUE THREAD

**Issue by b-per** [plietar/rust-protobuf-macros#7]
Hi,

I am trying to compile librespot on my Raspberry Pi 2 running Raspbian Jessie using the rust binaries from here: https://github.com/warricksothr/RustBuild and it fails with the following errors

``` /root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_init.rs:49:5:`

 49:20 error: mismatched types:
 expected `syntax::ptr::P<syntax::codemap::Spanned<syntax::ast::StmtKind>>`,
    found `syntax::codemap::Spanned<syntax::ast::StmtKind>`
(expected struct `syntax::ptr::P`,
    found struct `syntax::codemap::Spanned`) [E0308]
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_init.rs:49     cx.stmt_expr(e)
                                                                                                    ^~~~~~~~~~~~~~~
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_init.rs:49:5: 49:20 help: run `rustc --explain E0308` to see a detailed explanation
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_init.rs:73:28: 73:72 error: mismatched types:
 expected `syntax::codemap::Spanned<syntax::ast::StmtKind>`,
    found `syntax::ptr::P<syntax::codemap::Spanned<syntax::ast::StmtKind>>`
(expected struct `syntax::codemap::Spanned`,
    found struct `syntax::ptr::P`) [E0308]
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_init.rs:73                 stmts.push(emit_repeated(cx, sp, v, e_repeated.clone()))
                                                                                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_init.rs:73:28: 73:72 help: run `rustc --explain E0308` to see a detailed explanation
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_bind.rs:78:17: 78:48 error: mismatched types:
 expected `syntax::codemap::Spanned<syntax::ast::StmtKind>`,
    found `syntax::ptr::P<syntax::codemap::Spanned<syntax::ast::StmtKind>>`
(expected struct `syntax::codemap::Spanned`,
    found struct `syntax::ptr::P`) [E0308]
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_bind.rs:78                 cx.stmt_let_pat(sp, pat, value)
                                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/protobuf_bind.rs:78:17: 78:48 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to 3 previous errors`
```

My version of rustc is the following one: rustc 1.8.0-dev (f9543a9b7 2016-02-15)
Does it mean that the issue comes from rust-protobuf-macros ?

Regards,

**Comment by b-per** [comment]
Actually, there are also more errors before:

```
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/util.rs:44:31: 44:45 error: unresolved name `ast::DeclLocal` [E0425]
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/util.rs:44         let decl = respan(sp, ast::DeclLocal(local));
                                                                                                                     ^~~~~~~~~~~~~~
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/util.rs:44:31: 44:45 help: run `rustc --explain E0425` to see a detailed explanation
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/util.rs:45:22: 45:35 error: unresolved name `ast::StmtDecl` [E0425]
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/util.rs:45         P(respan(sp, ast::StmtDecl(P(decl), ast::DUMMY_NODE_ID)))
                                                                                                            ^~~~~~~~~~~~~
/root/.cargo/git/checkouts/rust-protobuf-macros-eb90ab0bc4dc65d2/master/src/util.rs:45:22: 45:35 help: run `rustc --explain E0425` to see a detailed explanation
```

**Comment by smoltron** [comment]
Same here. Does not compile and therefore cannot build librespot. My rustc is 1.8.0-dev (57c357d89 2016-02-16). What is the compatible rustc version?

**Comment by plietar** [comment]
Fixed by 37c6a34
